File: CS/PersonalOrganizer/ContactContextDataModel/ContactContextUnitOfWork.cs
using System;
using System.Linq;
using System.Data;
using System.Data.Entity;
using System.Linq.Expressions;
using System.Collections.Generic;
using PersonalOrganizer.Common.Utils;
using PersonalOrganizer.Common.DataModel;
using PersonalOrganizer.Common.DataModel.EntityFramework;
using PersonalOrganizer.Model;

namespace PersonalOrganizer.ContactContextDataModel {
    /// <summary>
    /// A ContactContextUnitOfWork instance that represents the run-time implementation of the IContactContextUnitOfWork interface.
    /// </summary>
    public class ContactContextUnitOfWork : DbUnitOfWork<ContactContext>, IContactContextUnitOfWork {

        public ContactContextUnitOfWork(Func<ContactContext> contextFactory)
            : base(contextFactory) {
        }

        IRepository<Contact, int> IContactContextUnitOfWork.Contacts {
            get { return GetRepository(x => x.Set<Contact>(), x => x.Id); }
        }
    }
}

File: VB/PersonalOrganizer/ContactContextDataModel/ContactContextUnitOfWork.vb
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports PersonalOrganizer.Common.Utils
Imports PersonalOrganizer.Common.DataModel
Imports PersonalOrganizer.Common.DataModel.EntityFramework
Imports PersonalOrganizer.Model

Namespace PersonalOrganizer.ContactContextDataModel
    ''' <summary>
    ''' A ContactContextUnitOfWork instance that represents the run-time implementation of the IContactContextUnitOfWork interface.
    ''' </summary>
    Public Class ContactContextUnitOfWork
        Inherits DbUnitOfWork(Of ContactContext)
        Implements IContactContextUnitOfWork

        Public Sub New(ByVal contextFactory As Func(Of ContactContext))
            MyBase.New(contextFactory)
        End Sub

        Private ReadOnly Property IContactContextUnitOfWork_Contacts() As IRepository(Of Contact, Integer) Implements IContactContextUnitOfWork.Contacts
            Get
                Return GetRepository(Function(x) x.Set(Of Contact)(), Function(x) x.Id)
            End Get
        End Property
    End Class
End Namespace
